trestle/utils/load_distributed.py

```diff
@@ -0,0 +1,120 @@
+# -*- mode:python; coding:utf-8 -*-
+
+# Copyright (c) 2020 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module to load distributed model."""
+
+from pathlib import Path
+from typing import Any, Dict, List, Tuple, Type, Union
+
+from trestle.core import utils
+from trestle.core.base_model import OscalBaseModel
+from trestle.utils import fs
+
+
+def load_list(filepath: Path) -> Tuple[OscalBaseModel, str, List[OscalBaseModel]]:
+    """Given path to a directory of list(array) models, load the distributed models."""
+    aliases_not_to_be_stripped = []
+    instances_to_be_merged: List[OscalBaseModel] = []
+    # TODO: FIXME: fs.get_stripped_contextual_model fails without absolute file path!!! FIX IT!!
+    collection_model_type, collection_model_alias = fs.get_stripped_contextual_model(filepath.absolute())
+
+    for path in sorted(Path.iterdir(filepath)):
+
+        # ASSUMPTION HERE: if it is a directory, there's a file that can not be decomposed further.
+        if path.is_dir():
+            continue
+        model_type, model_alias, model_instance = load_distributed(path)
+
+        instances_to_be_merged.append(model_instance)
+        aliases_not_to_be_stripped.append(model_alias.split('.')[-1])
+
+    return collection_model_type, collection_model_alias, instances_to_be_merged
+
+
+def load_dict(filepath: Path) -> Tuple[OscalBaseModel, str, Dict[str, OscalBaseModel]]:
+    """Given path to a directory of additionalProperty(dict) models, load the distributed models."""
+    model_dict: Dict[str, OscalBaseModel] = {}
+    collection_model_type, collection_model_alias = fs.get_stripped_contextual_model(filepath.absolute())
+    for path in sorted(Path.iterdir(filepath)):
+        model_type, model_alias, model_instance = load_distributed(path)
+        field_name = path.parts[-1].split('__')[0]
+        model_dict[field_name] = model_instance
+
+    return collection_model_type, collection_model_alias, model_dict
+
+
+def load_distributed(
+    file_path: Path,
+    collection_type: Type[Any] = None
```

Comment:
Going with Optional[Type[Any]]. Otherwise it's `Optional[Union[Type[List[OscalBaseModel]], Type[Dict[str, OscalBaseModel]]]]`, which looks horrible. Also I don't understand why mypy needs annotations for List items, when I'm using Type[List]. 😕